src/hazelcore/extractions/ocaml/OCamlExtraction_Exp.re

```diff
@@ -0,0 +1,308 @@
+/* Theorem: If extract(ctx, d) = Some(ce, ty) and ctx |- d : ty then ce :_ocaml Typ.extract(ty) */
+
+type t =
+  | OcamlExp(ocaml_exp, HTyp.t)
+  | ExtractionFailed(string)
+and ocaml_exp = string;
```

Comment:
You should pass an indentation level as argument so the code you produce is properly indented.

Reply:
We can do this, but it's also fine imo to just run it through an ocaml pretty printer at the end to avoid keeping track of indentation levels.

Reply:
I would prefer an OCaml pretty printer, but I wasn't able to find a pretty printer that was usable as a library and not an external program that the user would have to run.

Reply:
I would prefer using a pretty printer, but I couldn't find one that we could use as a library and not a separate program that users would have to run manually.

(Apologies if this message is a duplicate.  I thought I posted it, but can't find it now.)